public/docs/ts/latest/guide/index.jade

```diff
@@ -1,10 +1,501 @@
+  
+  The template, metadata, and component together describe the view.
+
+  We apply other metadata decorators in a similar fashion to guide Angular behavior. 
+  The `@Injectable`, `@Input`, `@Output`, `@RouterConfig` are a few of the more popular decorators
+  we'll master as our Angular knowledge grows.
+<br clear="all">
+:marked
+  The architectural take-away is that we must add metadata to our JavaScript code
+  so that Angular knows what to do.
+
+.l-main-section
+<a id="data-binding"></a>
+:marked
+  ## Data Binding
+  Without a framework, we would be responsible for pushing data values into the HTML controls and turning user responses
+  into actions and value updates. Writing such push/pull logic by hand is tedious, error-prone and a nightmare to
+  read as the experienced jQuery programmer can attest  
+figure
+  img(src="/resources/images/devguide/intro/databinding.png" alt="Data Binding" style="width:200px; float:left; margin-left:-40px;margin-right:10px" )
+:marked
+  Angular supports **data binding**, 
+  a mechanism for coordinating parts of a template with parts of a component.
+  We add binding markup to the template HTML to tell Angular how to connect both sides.
+  
+  There are four forms of data binding syntax. Each form has a direction - to the DOM, from the DOM, or in both directions -
+  as indicated by the arrows in the diagram.
+<br clear="all">
+:marked
+  We saw three forms of data binding in our [example](#template) template:
++makeExample('intro/ts/app/hero-list.component.html', 'binding')(format=".")
+:marked
+  * The <code>{&zwnj;{hero.name}}</code> "[interpolation](displaying-data.html#interpolation)" 
+  displays the component's `hero.name` property value within the `<div>` tags.
+  
+  * The `[hero]` [property binding](template-syntax.html#property-binding) passes the `selectedHero` from
+  the parent `HeroListComponent` to the `hero` property of the child `HeroDetailComponent`.
+  
+  * The `(click)` [event binding](user-input.html#click) calls the Component's `selectHero` method when the user clicks
+  on a hero's name
+
+  **Two-way data binding** is an important fourth form
+  that combines property and event binding in a single notation using the `ng-model` directive. 
+  We didn't have a two-way binding in the `HeroListComponent` template; here's an example from a different template:
+
++makeExample('intro/ts/app/hero-detail.component.html')(format=".")
+:marked
+  In two-way binding, a data property value flows to the input box from the component as with property binding.
+  The user's changes also flow back to the component, resetting the property to the latest value,
+  as with event binding.
+  
+  Angular processes *all* data bindings once per JavaScript event cycle, 
+  depth-first from the root of the application component tree. 
+figure
+  img(src="/resources/images/devguide/intro/component-databinding.png" alt="Data Binding" style="float:left; width:300px; margin-left:-40px;margin-right:10px" )
```

Comment:
I really like those graphics. This one here shows the connection between a component an the outer template where the component is used. It can be a bit confusing, because a component can have its own template/view and it's not entirely clear what template we have in this figure.

Maybe we should make clear that we have outer/inner template? "component template" doesn't work because everything is a component after all...